src/codec.rs

```diff
@@ -28,39 +28,75 @@ use core::{mem, slice};
 use arrayvec::ArrayVec;
 use core::marker::PhantomData;
 
+#[cfg(feature = "std")]
+use std::fmt;
+
+#[cfg_attr(feature = "std", derive(Debug))]
+#[derive(PartialEq)]
+pub struct Error(&'static str);
+
+impl Error {
```

Comment:
Could you implement `From<&'static str>` for `Error` please.

Reply:
ok done